### PR TITLE
🗃 Preserve location when reopening extension

### PR DESCRIPTION
**Pull request by henryboldi**
This also preserves the location's `state` attribute. For example, when the token clicked is passed to the single asset page.

After discovering `Router`'s `initialEntries` prop I was able to massively simplify the approach here.

⚠️ If a user navigates to a page they can't exit, there's no escape

### CODE REVIEW

**Review by wojciech-turek — DISMISSED** [review]
I have found a breaking bug. To replicate:

Go to Send page => Select data => Go to Sign Page =>  click away to close extension => go back to extension and click Reject

We are left with a blank screen, if i close and reopen the send page shows correctly

File: ui/pages/Popup.tsx [hunk 1]
```diff
@@ -24,9 +24,16 @@ const pagePreferences = Object.fromEntries(
   ])
 )
 
+const initialEntries = JSON.parse(
+  (localStorage.getItem("entries") !== "undefined" &&
```
Comment (greg-nagy) [inline ui/pages/Popup.tsx:28]: Not blocking but using `??` would have been simpler here.

File: ui/pages/Popup.tsx [hunk 2]
```diff
@@ -54,6 +61,39 @@ export default function Popup({ store }: { store: Store }): ReactElement {
...
+        "/keyring/"
+      )
+
+    // Initial extension load takes two renders because of setting
+    // animation control states. `initialEntries` needs to be a reversed
+    // version of route history entities. Without avoiding the initial load,
+    // entries will keep reversing.
+    if (renderCount.current > 1 && isNotOnKeyringRelatedPage) {
+      const entries = routeHistoryEntities
+        .reduce((agg: Partial<Location>[], entity) => {
```
Comment (greg-nagy) [inline ui/pages/Popup.tsx:80]: hm, did not know that agg also works. Thanks

**Review by greg-nagy — APPROVED** [review]
Love how easy this important feature was 

🥷

**Pull request merged**